staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -1375,6 +1383,29 @@ func mergeMap(original, patch map[string]interface{}, schema LookupPatchMeta, me
 	return original, nil
 }
 
+// discardNullValuesFromPatch discards all null values from patch.
+// It traverses for all slices and map types to discard all nulls.
+func discardNullValuesFromPatch(patchV interface{}) interface{} {
+	switch patchV.(type) {
+	case map[string]interface{}:
+		for k, v := range patchV.(map[string]interface{}) {
+			if v == nil {
+				delete(patchV.(map[string]interface{}), k)
+			} else {
+				discardNullValuesFromPatch(v)
+			}
+		}
+	case []interface{}:
+		patchS := patchV.([]interface{})
+		for i, v := range patchV.([]interface{}) {
+			patchS[i] = discardNullValuesFromPatch(v)
```

Comment:
I don't think this needs to reassign to `patchS[i]`:
```suggestion
			discardNullValuesFromPatch(v)
```